src/casts/interpret-casts-common.rkt

```diff
@@ -163,7 +161,9 @@ TODO write unit tests
       (cond$
        [(op=? t1 t2) t1]
        [(type-dyn?$ t1) t2]
-       [(type-dyn?$  t2) t1]
+       [(type-dyn?$ t2) t1]
+       [(or$ (Base-Type-Huh t1) (Base-Type-Huh t2))
+        (Error (Quote "inconsistent types"))]
```

Comment:
Isn't this case redundant?

Reply:
This check is important so that we access the ptr if it is indeed a ptr. Base types are just numbers.

Reply:
yes it does not.